From: Sergey Senozhatsky <senozhatsky@chromium.org>
Subject: zram: use GFP_KERNEL for post-processing slot allocation
Date: Thu, 5 Mar 2026 11:16:03 +0900

Post-processing slot allocation used to be performed under slot bit-lock,
so gfp was restricted, then we switched to a sleepable slot lock but gpf
still remained restricted.  Drop that restriction and use GFP_KERNEL
instead.

Link: https://lkml.kernel.org/r/20260305021619.617481-1-senozhatsky@chromium.org
Signed-off-by: Sergey Senozhatsky <senozhatsky@chromium.org>
Acked-by: Brian Geffon <bgeffon@google.com>
Cc: Jens Axboe <axboe@kernel.dk>
Cc: Minchan Kim <minchan@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 drivers/block/zram/zram_drv.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/block/zram/zram_drv.c~zram-use-gfp_kernel-for-post-processing-slot-allocation
+++ a/drivers/block/zram/zram_drv.c
@@ -297,7 +297,7 @@ static bool place_pp_slot(struct zram *z
 	struct zram_pp_slot *pps;
 	u32 bid;
 
-	pps = kmalloc_obj(*pps, GFP_NOIO | __GFP_NOWARN);
+	pps = kmalloc_obj(*pps, GFP_KERNEL | __GFP_NOWARN);
 	if (!pps)
 		return false;
 
_
